### DISCUSSION TITLE
Infinite redirect loop when signing in as end user to 1 app

### DISCUSSION THREAD

**Discussion by chrisshaw** [Budibase/budibase#1929] [Help] [answered]
Seems like a bug, but is it possible my hosting setup is to blame?

---

Firefox 89.0.2

---

- Droplet 1: Digital Ocean 1-Click Deploy, which I then updated to 9.7
![image](https://user-images.githubusercontent.com/2966918/124538098-0ff89580-dde9-11eb-8f08-6e26c7a5b38f.png)

- Droplet 2: Caddy reverse proxy
Super simple Caddyfile:
```
report.sidekick.education
reverse_proxy budibase_private_ip:10000
```

---

Not much is useful in the console. The client is hitting 403s when trying to fetch `ta_users`.
![image](https://user-images.githubusercontent.com/2966918/124539285-26075580-ddeb-11eb-8334-4118b67fd3f7.png)

The video I'm about to attach is wholly unhelpful, unfortunately. To clarify, the route oscillates between `https://report.sidekick.education/app_816bc33d60f74366be72552c62da5c88/` and adding a `#` before the final slash. So `https://report.sidekick.education/app_816bc33d60f74366be72552c62da5c88#/`

https://user-images.githubusercontent.com/2966918/124538184-2f8fbe00-dde9-11eb-9e9e-4721f4939df1.mp4

**Comment by aptkingston** [comment, marked as answer]
Hey @chrisshaw. You're correct that this is currently a bug. There's an issue already created (https://github.com/Budibase/budibase/issues/1862) so you can keep track of it there. 

Appreciate the detailed write up, that's really helpful.

**Reply by chrisshaw** [reply]
Awesome 👍

I see the issue's root cause was found 6 days ago, but it has not been assigned and it's obviously not in 9.7. Do you have a sense of progress on this issue? (Hate to be pushy but was hoping for go-live today.)

**Reply by aptkingston** [reply]
I'll have this fixed today, but just discussing internally at the moment what the best way to get this released is. I'll post an update here when I know.

**Comment by aptkingston** [comment]
Just on another note - I think the reason this might be happening is that the user signing in doesn't have a role set in the app they're trying to access. You might be able to fix this by making sure they have any sort of role set (even a basic role).

You can check this in the admin portal, by going to "Users" and selecting the user, then making sure they have a role for that app. They should have a role listed in this table for the app they want to access:

![image](https://user-images.githubusercontent.com/9075550/124599839-08bf9f00-de5e-11eb-9a68-71cb201cf47a.png)

Let me know if this issue still persists after they have a role, or if they already had a role.

**Reply by chrisshaw** [reply]
:dart: Thanks! Sure enough that seems to have calmed down the `ta_users` 403s.

> Now I'm not getting anything to render, which is probably just a mismatch with my RBAC. I'll tinker before I come back to cry for help.

**Reply by aptkingston** [reply]
Awesome! Yep not having a role set is the cause of the `ta_users` 403, which in turn causes a page reload because it thinks the user's token has expired, then the app will try to fetch the `ta_users` definition again on page load, causing a 403, and the cycle continues! So while setting a role fixes that issue, we need to improve the logic there.

**Comment by aptkingston** [comment]
@chrisshaw v0.9.71 has been released and should be generally available shortly, which fixes this.
